trace-sdk/src/test/java/io/bitrise/trace/data/management/formatter/cpu/SystemCpuDataFormatterTest.java

```diff
@@ -0,0 +1,111 @@
+package io.bitrise.trace.data.management.formatter.cpu;
+
+import androidx.annotation.NonNull;
+
+import com.google.protobuf.Timestamp;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import io.bitrise.trace.data.collector.cpu.CpuUsageData;
+import io.bitrise.trace.data.collector.cpu.SystemCpuUsageDataCollector;
+import io.bitrise.trace.data.dto.Data;
+import io.bitrise.trace.data.dto.FormattedData;
+import io.bitrise.trace.data.management.formatter.BaseDataFormatterTest;
+import io.opencensus.proto.metrics.v1.LabelKey;
+import io.opencensus.proto.metrics.v1.LabelValue;
+import io.opencensus.proto.metrics.v1.Metric;
+import io.opencensus.proto.metrics.v1.MetricDescriptor;
+import io.opencensus.proto.metrics.v1.Point;
+import io.opencensus.proto.metrics.v1.TimeSeries;
+
+import static io.bitrise.trace.data.dto.DataValues.cpu;
+import static io.bitrise.trace.data.dto.DataValues.getName;
+import static io.bitrise.trace.data.dto.DataValues.idle;
+import static io.bitrise.trace.data.dto.DataValues.ioWait;
+import static io.bitrise.trace.data.dto.DataValues.irq;
+import static io.bitrise.trace.data.dto.DataValues.nice;
+import static io.bitrise.trace.data.dto.DataValues.pct;
+import static io.bitrise.trace.data.dto.DataValues.percent;
+import static io.bitrise.trace.data.dto.DataValues.softIrq;
+import static io.bitrise.trace.data.dto.DataValues.state;
+import static io.bitrise.trace.data.dto.DataValues.steal;
+import static io.bitrise.trace.data.dto.DataValues.system;
+import static io.bitrise.trace.data.dto.DataValues.user;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Unit tests for {@link SystemCpuDataFormatter}.
+ */
+public class SystemCpuDataFormatterTest extends BaseDataFormatterTest {
+
+    final SystemCpuDataFormatter formatter = new SystemCpuDataFormatter();
+    final CpuUsageData.CpuStat cpuStat = new CpuUsageData.CpuStat(1,2,3,4,5,6,7,8);
+    final String metricDescriptionDescription = "System CPU Usage";
+
+    @Test
+    public void formatData_notCpuStat() {
+        final Data data = new Data(SystemCpuUsageDataCollector.class);
+        data.setContent("cpu usage");
+        assertArrayEquals(new FormattedData[]{}, formatter.formatData(data));
+    }
+
+    @Test
+    public void formatData() {
+        final Data data = new Data(SystemCpuUsageDataCollector.class);
+        data.setContent(cpuStat);
+
+        final FormattedData formattedData = formatter.formatData(data)[0];
+
+        final Timestamp timestamp = formattedData.getMetricEntity().getMetric()
+                .getTimeseries(0).getPoints(0).getTimestamp();
+
+        List<TimeSeries> expectedTimeSeries = new ArrayList<>(Arrays.asList(
```

Comment:
We can avoid this code duplication if the original is made package private in SystemCpuDataFormatter (with VisibleForTesting annotation), and we use it here too.